e2e-tests/src/main.rs

```diff
@@ -33,6 +34,53 @@ fn init_env() {
     env_logger::init();
 }
 
+/// Runs default test cases in sequence.
+fn run_default_test_cases(possible_test_cases: HashMap<&str, TestCase>, config: &Config) -> anyhow::Result<()> {
+    let _ = possible_test_cases
+        .iter()
+        .map::<anyhow::Result<()>, _> (
+            |(test_name, &test_case)| {
+                run_test_case(test_case, test_name, config)?;
+                Ok(())
+            }
+        ).collect::<Vec<_>>();
+    Ok(())
+}
+
+/// Runs specified test cases in sequence.
+/// Checks whether each provided test case is valid.
+fn run_specified_test_cases(test_cases: Vec<String>, possible_test_cases: HashMap<&str, TestCase>, config: &Config) -> anyhow::Result<()> {
```

Comment:
nit: a really long line. You should run `cargo fmt` in the e2e-tests directory (and we need to add this to pipelines&checks but not in this pr).

Reply:
Right, forgot to run a local pipeline which does this.